src/components/ChangePassword/ChangePassword.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import { CardText, CardBody, CardSubtitle, FormGroup } from 'reactstrap';
+import { connect } from 'react-redux';
+import queryString from 'query-string';
+import propTypes from 'prop-types';
+import './changePassword.scss';
+import { withRouter } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import * as actions from '../../../store/actions/authActions/changePassword';
+import Card from '../Card/Card';
+import Button from '../Button';
+import Input from '../Inputs/Input';
+import Footer from '../Footer/Footer';
+import { Header } from '../Header/Header';
+import { Heading } from '../Heading/Heading';
+
+/**
+ * @description
+ * @param
+ * @returns
+ */
+export class ChangePassword extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    this.state = {
+      password: '',
+      confirmPassword: '',
+      error: '',
+      passwordFieldError: '',
+      confirmPasswordFieldError: '',
+      success: '',
+    };
+  }
+
+  handleInput(e) {
+    const { value, name } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  async handleSubmit() {
+    const { password, confirmPassword } = this.state;
+    const { changePassword, history, location } = this.props;
+    this.setState({ passwordFieldError: '' });
+    this.setState({ confirmPasswordFieldError: '' });
+    this.setState({ error: '' });
+    this.setState({ success: '' });
+    if (password === '') {
+      this.setState({ passwordFieldError: 'password field cannot be empty' });
+    } else if (password === confirmPassword) {
+      const { token } = queryString.parse(location.search);
+      await changePassword(password, token, history);
+      const { passwordChangeSuccess, passwordChangeError } = this.props;
+      this.setState({ error: passwordChangeError });
+      this.setState({ password: '', confirmPassword: '' });
+      this.setState({ success: passwordChangeSuccess });
```

Comment:
I still don't understand why we can't handle all of this in one single `this.setState()`? @beejay1293